### ISSUE TITLE
(cpp) Regression between 9.10.0 and 9.11.0, "class kernel" is not highlighted inside lambda

### ISSUE THREAD

**Issue by andykras** [highlightjs/highlight.js#3103]
**Describe the issue**
"class kernel" is not highlighted inside lambda

**Which language seems to have the issue?**
`cpp`

**Are you using `highlight` or `highlightAuto`?**
Both highlight and highlightBlock gives the same result.
Also tested against 10.7.1 with highlightElement

...

**Sample Code to Reproduce**
http://jsfiddle.net/andykras/wpdcv9q0/15  ```Highlight.js 9.11.0 NOT worked```

http://jsfiddle.net/andykras/9mvuaybz/22  ```Highlight.js 9.10.0 worked```

```cpp
class kernel_br{};

q.submit([&](handler &cgh)
{
  // here <class kernel_br> should be highlighted
  cgh.parallel_for<class kernel_br>(workRange, reduction(sumAcc, 0));
});

cgh.parallel_for<class kernel_br>(workRange, reduction(sumAcc, 0));
```
**Expected behavior**
"class kernel" should be highlighted the same way it's in lambda or not.

**Additional context**
This is a C++ SYCL kernels, check more code like this here https://developer.codeplay.com/products/computecpp/ce/guides/sycl-guide/multiple-kernels#multiple-kernels

**Comment by joshgoebel** [comment]
I'll look into this after #3078 is merged.

**Comment by joshgoebel** [comment]
Seems resolved in `main` already, likely by https://github.com/highlightjs/highlight.js/pull/3078.

**Comment by andykras** [comment]
> Seems resolved in `main` already, likely by #3078.

Thank you!
